src/library/VaultLib.sol

```diff
@@ -91,6 +91,11 @@ library VaultLib {
             revert IVault.InvalidNativeAssetDecimals(decimals_);
         }
 
+        // Vault does not support assets with more than 18 decimals
+        if (decimals_ > 18) {
```

Comment:
Should this be `decimals_ > vaultStorage.decimals` as in no asset has more decimals than the vault instead?

That does imply if you have a vault with 8 decimals like ynBfBTCk you can't add an asset that has 18 decimals like BTCB